ignite-base/App/Containers/ListviewSearchingExample.js

```diff
@@ -0,0 +1,95 @@
+// @flow
+
+import React from 'react'
+import { View, Text, ListView } from 'react-native'
+import { connect } from 'react-redux'
+// For empty lists
+import AlertMessage from '../Components/AlertMessage'
+// Styles
+import styles from './Styles/ListviewExampleStyle'
+
+class ListviewExample extends React.Component {
+  state: {
+    dataSource: Object
+  }
+
+  constructor (props) {
+    super(props)
+    /* ***********************************************************
+    * Teach datasource how to detect if rows are different
+    * Make this function fast!  Perhaps something like:
+    *   (r1, r2) => r1.id !== r2.id}
+    *************************************************************/
+    const rowHasChanged = (r1, r2) => r1 !== r2
+
+    // DataSource configured
+    const ds = new ListView.DataSource({rowHasChanged})
+
+    // Datasource is always in state
+    this.state = {
+      dataSource: ds.cloneWithRows(props.results)
+    }
+  }
+
+  /* ***********************************************************
+  * `_renderRow` function -How each cell/row should be rendered
+  * It's our best practice to place a single component here:
+  *
+  * e.g.
+    return <MyCustomCell title={rowData.title} description={rowData.description} />
+  *************************************************************/
+  _renderRow (searchTerm) {
+    return (
+      <View style={styles.row}>
+        <Text style={styles.boldLabel}>{searchTerm}</Text>
+      </View>
+    )
+  }
+
+  /* ***********************************************************
+  * If your datasource is driven by Redux, you'll need to
+  * reset it when new data arrives.
+  * DO NOT! place `cloneWithRows` inside of render, since render
+  * is called very often, and should remain fast!  Just replace
+  * state's datasource on newProps.
+  *
+  * e.g.
+  *************************************************************/
+  componentWillReceiveProps (newProps) {
+    if (newProps.results) {
+      this.setState({
+        dataSource: this.state.dataSource.cloneWithRows(newProps.results)
+      })
+    }
+  }
+
+  // Used for friendly AlertMessage
+  // returns true if the dataSource is empty
+  _noRowData () {
```

Comment:
I'm not a fan of `_`.  :(    Again, just general bitching.  Nothing really do with this PR.  Hah... oh boy.  I'm annoying myself now.